### PR TITLE
Domains: Update Failed Purchase page to include successful ones

### PR DESCRIPTION
**Pull request by aidvu**
- Show successful purchases under failed ones
- Refactor code to use the same pattern as all checkout thank you components

Test:
1. Add domain to cart
2. Add site redirect to cart
3. Make domain purchase fail on backend

Before:
failed purchases

After:
new failed after

### CODE REVIEW

File: client/my-sites/upgrades/checkout-thank-you/header.jsx [hunk 1]
```diff
@@ -150,7 +159,8 @@ class CheckoutThankYouHeader extends React.Component {
 
 CheckoutThankYouHeader.propTypes = {
 	isDataLoaded: React.PropTypes.bool.isRequired,
-	primaryPurchase: React.PropTypes.object
+	primaryPurchase: React.PropTypes.object,
+	failedPurchases: React.PropTypes.array
```
Comment (klimeryk) [inline client/my-sites/upgrades/checkout-thank-you/header.jsx]: Seems you could just pass something like `hasFailedPurchases` (boolean equal to `failedPurchases.length > 0`) and simplify the code a bit :)

Comment (aidvu) [inline client/my-sites/upgrades/checkout-thank-you/header.jsx]: Hmm, yeah, overengineered a bit. It's not being used in there. Will do, and then merge.

**Comment by ranh** [comment]
Does this apply only to domains @aidvu?

Anyway, here's a suggestion for different copy and a different layout (putting the successful items first):

> **Some items failed.**
> 
> Some of the items in your cart could not be added.
> 
> ----
> 
> These items were added successfully:
> 
>  - item.
>  - item.
> 
> ----
> 
> These items could not be added:
> 
> - item.
> - item.
> 
>  We added credits to your account, so you can try adding these items again. If the problem persists, please don’t hesitate to [contact support](#). 
>

**Comment by aidvu** [comment]
@ranh Applies for any failed purchase. Will do a new PR to update the copy. Thank you!

**Pull request merged**